src/renderer/pages/Project.tsx

```diff
@@ -1,28 +1,81 @@
-import { Stack } from '@mui/material';
-import { Box } from '@mui/system';
+import { Box, Stack } from '@mui/material';
+import { useCallback, useEffect } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
 import StandardButton from 'renderer/components/StandardButton';
-import { projectOpened } from 'renderer/store/actions';
+import TranscriptionBlock from 'renderer/components/TranscriptionBlock';
+import { transcriptionCreated } from 'renderer/store/actions';
+import { Transcription } from 'sharedTypes';
 import { ApplicationStore } from '../store/helpers';
 
+/*
+This is the page that gets displayed while you are editing a video.
+It will be primarily composed of the transcription area, an editable text box whose
+changes get reflected in the video. In addition to that, there is a video preview
+section to the side among other things.
+*/
 const ProjectPage = () => {
+  const dispatch = useDispatch();
   const currentProject = useSelector(
     (store: ApplicationStore) => store.currentProject
   );
 
-  const dispatch = useDispatch();
+  // RK: I really shouldn't use transcriptionCreated here - but i'm lazy and it works
+  const saveTranscription: (transcription: Transcription) => void = useCallback(
+    (transcription) => dispatch(transcriptionCreated(transcription)),
+    [dispatch]
+  );
+
+  const deleteWord = (firstWordIndex: number, numberOfWords: number) => {
+    if (currentProject && currentProject.transcription) {
+      // eslint-disable-next-line no-plusplus
+      for (let i = firstWordIndex; i < firstWordIndex + numberOfWords; i++) {
+        currentProject.transcription.words[i].deleted = true;
+      }
+      saveTranscription(currentProject.transcription);
+    }
+  };
 
-  if (currentProject === null) {
+  const deleteText = async () => {
+    const highlightedWords = window.getSelection();
+    if (
+      highlightedWords?.anchorNode?.parentElement?.dataset.type === 'word' &&
+      highlightedWords?.focusNode?.parentElement?.dataset.type === 'word'
+    ) {
+      const anchor = await Number(
+        highlightedWords?.anchorNode?.parentElement?.dataset.index
+      );
+      const focus = await Number(
+        highlightedWords?.focusNode?.parentElement?.dataset.index
+      );
+
+      const start = Math.min(anchor, focus);
+      const end = Math.max(anchor, focus);
+      deleteWord(start, end - start + 1);
+    }
+  };
+
+  const onKeyDown = (event: KeyboardEvent) => {
+    if (event.key === 'Delete' || event.key === 'Backspace') {
+      deleteText();
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener('keydown', onKeyDown);
+
+    return () => {
+      window.removeEventListener('keydown', onKeyDown);
+    };
+  });
```

Comment:
This use effect will currently add the event listener on every render and remove it at "end" of render. 
Just checking that this was the intention?

If you just wanted to add the event listener when the page loads, and remove it when the page closes, then add an empty dependency array after the function argument.

Reply:
Ooh good catch on that one hugh

Reply:
linter isn't happy with the use of an empty dependency array :'( - will look into it further in another PR

Reply:
Yeah the current impl shouldnt cause any bugs it's just inefficient